### PR TITLE
GDB cheatsheet added

### PR DESCRIPTION
**Pull request by harisphnx**
What does your Instant Answer do?
Its a cheatsheet for the Basics commands of GDB.

What problem does your Instant Answer solve (Why is it better than organic links)?
It gives access to some of the basic commands of GDB. Commands for running a program, setting breakpoints & watchpoints, for execution control, working with files and so on.

What is the data source for your Instant Answer? (Provide a link if possible)
http://users.ece.utexas.edu/~adnan/gdb-refcard.pdf

Why did you choose this data source?
Because this contains a good collection of gdb commands and their description.

Are there any other alternative (better) data sources?
I didn't need any. This was enough.

What are some example queries that trigger this Instant Answer?
gdb cheatsheet OR gnu debugger cheatsheet. I can add more aliases if needed.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)
Programmers and testers.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?
https://duck.co/ideas/idea/6515/gdb-cheatsheet

Which existing Instant Answers will this one supercede/overlap with?
I don't think one exists which this will be related to.

Are you having any problems? Do you need our help with anything?
No, Just need it reviewed

Where did you hear about DuckDuckHack? (For first time contributors)
The internet

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

screenshot from 2015-12-16 21 42 12
---
https://duck.co/ia/view/gdb_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata post coming soon

**Comment by codenirvana** [comment]
@harisphnx Don't know how but you pushed ```node_modules``` folder with this PR.
Please fix this issue first!

**Comment by harisphnx** [comment]
@codenirvana I am not well versed with git. I followed steps and did this. I don't really know how to fix whatever issue you have mentioned.

**Comment by codenirvana** [comment]
@harisphnx Check [this](https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1912/files)
You are using codio, right? How you get all that ```node_modules``` stuff?

You can try this to fix this issue:
```git rm -rf node_modules
git commit -m 'deleted node_modules'```

File: share/goodie/cheat_sheets/json/gdb.json [hunk 1]
```diff
@@ -0,0 +1,184 @@
+{
+    "id": "gdb_cheat_sheet",
+    "name": "GNU Debugger",
+    "description": "Basic GDB commands",
+    "metadata": {
+        "sourceName": "GDB refcard",
+        "sourceUrl": "http://users.ece.utexas.edu/~adnan/gdb-refcard.pdf"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/gdb.json]: @harisphnx lets change source name to  "UTEXAS"

**Comment by harisphnx** [comment]
@jophab I was a little confused about that source name too. Changed now.

File: share/goodie/cheat_sheets/json/gdb.json [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        "Breakpoints & Watchpoints",
+        "Program Stack",
+        "Execution Control",
+        "Working Files",
+        "Miscellaneous"
+    ],
+    "sections": {
+        "Starting GDB": [
+            {
+                "val": "Starting GDB, with no debugging files",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/gdb.json]: @harisphnx I this comma needed?

Comment (harisphnx) [inline share/goodie/cheat_sheets/json/gdb.json]: I don't think I followed the *pair of brackets* thing anywhere for the "val". I can put a semi-colon though, to introduce a connecting a effect like I did for *backtrack* or *frame*.

Comment (jophab) [inline share/goodie/cheat_sheets/json/gdb.json]: @harisphnx  Lets change this to "Start GDB with no debugging files"
Isnt it better?

Comment (harisphnx) [inline share/goodie/cheat_sheets/json/gdb.json]: @jophab I guess so. I was also confused. ok.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=GDB Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/gdb_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@harisphnx sorry for the delay, this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:).

**Comment by harisphnx** [comment]
@moollaza You're welcome. I was thinking of moving a step forward. Is there any other project other then cheatsheet where I can contribute. I mean i know about spice and all, but anything else that would suit my level of knowledge.

**Pull request merged**